### PR TITLE
Update dependencies for v1 branch 

### PR DESCRIPTION
**Pull request by satk0**

### CODE REVIEW

File: .github/workflows/go-application.yml [hunk 1]
```diff
@@ -21,12 +21,12 @@ jobs:
     runs-on: ubuntu-20.04
     steps:
       - name: Checkout code
-        uses: actions/checkout@v4
+        uses: actions/checkout@v4.1.1
```
Comment (kuskoman) [inline .github/workflows/go-application.yml:24]: don't pin the version

**Review by kuskoman — CHANGES_REQUESTED** [review]

**Comment by kuskoman** [comment]
@satk0 prometheus tests are failing

**Comment by kuskoman** [comment]
@satk0 seems like the upgrade-dependencies script differs between v1 and v2
also, you may want to check this pr #280

**Comment by satk0** [comment]
@kuskoman Thanks a ton for all the hints. I am giving it another try 😎

**Review by kuskoman — APPROVED** [review]

**Pull request merged**